### PR TITLE
Remove unnecessary factory layer

### PR DESCRIPTION
**Pull request by mrkm4ntr**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
`podLister` field of struct `PredicateMetadataFactory` is no longer used.  So we can remove `PredicateMetadataFactory` and get `PredicateMetadataProducer` directly.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test
/priority backlog

File: pkg/scheduler/algorithm/predicates/metadata.go [hunk 1]
```diff
@@ -190,16 +184,8 @@ func RegisterPredicateMetadataProducerWithExtendedResourceOptions(ignoredExtende
...
-func NewPredicateMetadataFactory(podLister algorithm.PodLister) PredicateMetadataProducer {
-	factory := &PredicateMetadataFactory{
-		podLister,
-	}
-	return factory.GetMetadata
-}
-
-// GetMetadata returns the predicateMetadata used which will be used by various predicates.
-func (pfactory *PredicateMetadataFactory) GetMetadata(pod *v1.Pod, nodeNameToInfoMap map[string]*schedulernodeinfo.NodeInfo) PredicateMetadata {
+// GetPredicateMetadata returns the predicateMetadata used which will be used by various predicates.
```
Comment (bsalamat) [inline pkg/scheduler/algorithm/predicates/metadata.go]: s/the predicateMetadata used which/the predicateMetadata which/

**Review by bsalamat — COMMENTED** [review]
Looks good. Just a minor comment.

**Comment by mrkm4ntr** [comment]
@bsalamat Fixed, PTAL.

**Review by draveness — COMMENTED** [review]
/lgtm

Thanks @mrkm4ntr

**Review by bsalamat — COMMENTED** [review]
/lgtm
/approve

**Comment by mrkm4ntr** [comment]
@bsalamat k8s-ci-robot didn't work. Please approve again.

**Review by bsalamat — COMMENTED** [review]
/approve

**Comment by liggitt** [comment]
/hold

per [release-managers discussion](https://kubernetes.slack.com/archives/CJH2GBF7Y/p1567084773002200), holding low-priority, non-milestone-targeted PRs in order to free merge pool bandwidth for 1.16-targeted PRs

feel free to unhold once there are no [lgtm+approve+green 1.16-targeted PRs](https://github.com/kubernetes/kubernetes/pulls?utf8=✓&q=is%3Apr+is%3Aopen+label%3Algtm+label%3Aapproved+milestone%3Av1.16+status%3Asuccess) remaining.

**Pull request merged**